python/ctsm/subset_data.py

```diff
@@ -0,0 +1,556 @@
+        type=plat_type,
+        default=15,
+    )
+    rg_parser.add_argument(
+        "--lon1",
+        help="Region start longitude. [default: %(default)s]",
+        action="store",
+        dest="lon1",
+        required=False,
+        type=plon_type,
+        default=275.0,
+    )
+    rg_parser.add_argument(
+        "--lon2",
+        help="Region end longitude. [default: %(default)s]",
+        action="store",
+        dest="lon2",
+        required=False,
+        type=plon_type,
+        default=330.0,
+    )
+    rg_parser.add_argument(
+        "--reg",
+        help="Region name or tag. [default: %(default)s]",
+        action="store",
+        dest="reg_name",
+        required=False,
+        type=str,
+        default="",
+    )
+    rg_parser.add_argument(
+        "--create-mesh",
+        help="Subset a mesh file for a region.",
+        action="store_true",
+        dest="create_mesh",
+        required=False,
+        default=False,
+    )
+
+    # -- common options between both subparsers
+    for subparser in [pt_parser, rg_parser]:
+        subparser.add_argument(
+            "--create-domain",
+            help="Create CLM domain file at single point/region.",
+            action="store_true",
+            dest="create_domain",
+            required=False,
+        )
+        subparser.add_argument(
+            "--create-surface",
+            help="Create surface data file at single point/region.",
+            action="store_true",
+            dest="create_surfdata",
+            required=False,
+        )
+        subparser.add_argument(
+            "--create-landuse",
+            help="Create landuse data file at single point/region.",
+            action="store_true",
+            dest="create_landuse",
+            required=False,
+        )
+        subparser.add_argument(
+            "--create-datm",
+            help="Create DATM forcing data at single point/region.",
+            action="store_true",
+            dest="create_datm",
+            required=False,
+        )
+        subparser.add_argument(
+            "--create-user-mods",
+            help="Create user mods directories and files for running CTSM with the subset data.",
+            action="store_true",
+            dest="create_user_mods",
+            required=False,
+        )
+        subparser.add_argument(
+            "--datm-syr",
+            help="Start year for creating DATM forcing at single point/region. [default: %("
+                 "default)s]",
+            action="store",
+            dest="datm_syr",
+            required=False,
+            type=int,
+            default=1901,
+        )
+        subparser.add_argument(
+            "--datm-eyr",
+            help="End year for creating DATM forcing at single point/region. "
+                 "[default: %(default)s]",
+            action="store",
+            dest="datm_eyr",
+            required=False,
+            type=int,
+            default=2014,
+        )
```

Comment:
Hi @wwieder Thanks for your comment. I am not sure what you mean here exactly. Do you think this would be in scope of subset_data developments? If so, we need to have a discussion to clear this out.

Reply:
@wwieder we think you are really talking about the settings for the case itself. That will show up in the user-mod directory that's created and the user can change it there. So I don't think this is part of what's being done here. But, I think we should talk about this more after we bring subset_data in and we can show you how it works. So I think is something possibly for future discussion. Or it might just be cleared up when we demonstrate how this all works.

So for this PR I'm taking this as handled.